ddp-workspace/projects/ddp-dsm-ui/src/app/participant-list/participant-list.component.ts

```diff
@@ -1465,179 +1470,37 @@ export class ParticipantListComponent implements OnInit {
   }
 
   public isSortField(name: string): boolean {
-    return name === this.sortField;
+    if (this.sortBy) {
+      return name === this.sortBy.innerProperty;
+    } else {
+      return false;
+    }
   }
 
   sortByColumnName(col: Filter, sortParent: string): void {
-    this.sortDir = this.sortField === col.participantColumn.name ? (this.sortDir === 'asc' ? 'desc' : 'asc') : 'asc';
-    this.sortField = col.participantColumn.name;
+    this.sortOrder = this.sortBy !== null
+          ? this.sortBy.innerProperty === col.participantColumn.name
+          ? ( this.sortOrder === 'asc' ? 'desc' : 'asc' ) : 'asc' : 'asc';
```

Comment:
Can you please double-check whether the condition is correct ? (lines 1481-1483)

Reply:
why did you get doubts about it?

Reply:
It is quite hard to read (multiple conditions with ternary operators) and maintain if needed.
High possibility to make a mistake.

Reply:
that is better, thank you!